one-ui/src/main/java/com/marklogic/hub/oneui/models/HubConfigSession.java

```diff
@@ -1000,14 +1002,14 @@ public void afterPropertiesSet() throws Exception {
         if (hubConfigImpl.getAdminManager() == null) {
             hubConfigImpl.setAdminManager(new AdminManager());
         }
+        String projectDirectory = environmentService.getProjectDirectory();
```

Comment:
After checking the HubConfigImpl, I think we can simplify unnecessary calls here. We even do not need to check if the hubConfigImpl.getAppConfig() is null because it is null for sure.

HubConfigImpl.java 
------------------------------------
    protected void hydrateAppConfigs(Properties properties) {
        com.marklogic.mgmt.util.PropertySource propertySource = properties::getProperty;

        if (appConfig != null) {
            // Still need to call this since the setter also "updates" appConfig with DHF-specific values
            setAppConfig(appConfig);
        } else {
            DefaultAppConfigFactory appConfigFactory = new DefaultAppConfigFactory(propertySource);
            if (hubProject != null && !StringUtils.isEmpty(hubProject.getProjectDirString())) {
                appConfigFactory.setProjectDir(hubProject.getProjectDir().toFile());
            }
            setAppConfig(appConfigFactory.newAppConfig());
        }
  }
...

after changing the HubConfigImpl class, we can only have the following three lines' codes

hubConfigImpl.createProject(projectDirectory);
hubConfigImpl.refreshProject();
this.dataHub = new DataHubImpl(this);

Reply:
@hao1st Are you saying to change hydrateAppConfigs? I know for certain that AppConfig is not always null in the context of that method.

Reply:
Yes, I think the root cause is how we glue those HubConfig/HubProject/AppConfig together. For one-ui installation case, we only need to deal with the situation when AppConfig is null.  And my gut feeling is that passing a valid project directory to AppConfig is always what we expect.  Plus I think the one-ui higher level api call should be simplier during createProject/refreshProject. For me, the following calls are confused and a hacker way to workaround.
```
public void afterPropertiesSet() in HubConfigSession class

        if (hubConfigImpl.getAppConfig() == null) {
            hubConfigImpl.setAppConfig(new AppConfig(), true);
            hubConfigImpl.setAppConfig(new AppConfig(Paths.get(projectDirectory).toFile()), true);
        }
        hubConfigImpl.createProject(environmentService.getProjectDirectory());
        hubConfigImpl.createProject(projectDirectory);
        hubConfigImpl.refreshProject();
        // resetting app config to trigger updates to values
        hubConfigImpl.setAppConfig(hubConfigImpl.getAppConfig(), false);
        hubConfigImpl.hydrateConfigs();
```